app/js/helpers/jsonHelper.js

```diff
@@ -0,0 +1,92 @@
+export class JSONHelper {
+  composeJson(searchParameters) {
+    const query = {};
+    query.type = "org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition";
+    query.columns = this.addColumnsToDisplay();
+    let counter = 0;
+    query.rowFilters = [];
+    for(let field in searchParameters) {
+      if(this.isNullValues(searchParameters[field])) {
+        continue;
+      }
+      if(searchParameters[field] != 'all' && searchParameters != '') {
+        query.rowFilters[counter] = {};
+        query.rowFilters[counter].key = this.getDefinitionLibraryKey(field, searchParameters[field]);
+      }
+      if(Array.isArray(searchParameters[field])) {
+        query.rowFilters[counter].parameterValues = this.getParameterValues(searchParameters[field]);
+      }
+      query.rowFilters[counter].type = "org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition";
+      counter += 1;
+    }
+    query.customRowFilterCombination = this.composeFilterCombination(query.rowFilters);
+    return query;
+  }
+
+  isNullValues(fieldValues) {
+    if(Array.isArray(fieldValues)) {
+      return (!fieldValues[0].value) ? true : false;
+    }
+    return (fieldValues === 'all' || !fieldValues) ? true : false;
```

Comment:
Not in all cases, this was particularly used for some select field that has an option to depict "all" options e.g for the gender field, to select both male and female you choose the "all" option